docs/02-app/01-building-your-application/06-optimizing/04-metadata.mdx

```diff
@@ -57,7 +57,7 @@ type Props = {
 
 export async function generateMetadata(
   { params, searchParams }: Props,
-  parent?: ResolvingMetadata
+  parent: ResolvingMetadata
```

Comment:
I suggest keeping it and changing the code below: (await parent)?.openGraph?.images
The reason is parent will be undefined if called from a segment that doesn't have a parent.

> 


```suggestion
  parent?: ResolvingMetadata
```

Reply:
It appears that the second argument of `generateMetadata` is expected to be of type `ResolvingMetadata` rather than `ResolvingMetadata | undefined`.

ref: https://github.com/vercel/next.js/blob/canary/packages/next/src/build/webpack/plugins/next-types-plugin/index.ts#L123